### PR TITLE
KFSPTS-13973 upgrade to the 11/29/18 version of financials

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Comment by cah292** [comment]
I'm still reviewing the code, but I did have one question so far. I noticed that our receiptProcessing-specific implementation of CsvBatchInputFileTypeBase has been removed by these changes. However, it seems that our implementation specifically customized the CSVReader setup to change the escape character to "|", which might not be present in the base KFS version of this class. It looks like such a character override was added a few years ago for KFSPTS-4136 and KFSPTS-4104, as part of some work related to ImageNow processing.

Do you happen to know whether we still need this custom character handling in the ImageNow CSV files, or who we could contact to find out?

**Comment by se12** [comment]
Chad,

As far as ImageNow questions, you should direct them to Gary Tidd (gmt73) and Susan Li (xl58)

File: src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurchaseOrderValidation.xml [hunk 1]
```diff
@@ -25,9 +25,7 @@
   		<property name="validations">
   			<list>  				
  				<!-- KFSPTS-2096 -->
-				<bean parent="Purchasing-saveItemValidation" scope="prototype">								
-					<property name="quitOnFail" value="false" />
-				</bean>  										
+				<bean parent="Purchasing-saveItemValidation" scope="prototype" p:quitOnFail="true" />
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurchaseOrderValidation.xml]: The old XML was setting "quitOnFail" to false, whereas this revised XML is setting "quitOnFail" to true. Can you please confirm whether the new behavior is the correct option?

File: src/main/java/org/kuali/kfs/coa/businessobject/Account.java [hunk 2]
```diff
@@ -171,13 +171,14 @@ public Account() {
      */
     public List<PersistableBusinessObject> generateDeactivationsToPersist() {
         // Retrieve all the existing sub accounts for this
-        List<SubAccount> bosToDeactivate = new ArrayList<SubAccount>();
+        List<SubAccount> bosToDeactivate = new ArrayList ();
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/org/kuali/kfs/coa/businessobject/Account.java]: Will add

File: src/main/java/org/kuali/kfs/coa/businessobject/Account.java [hunk 3]
```diff
@@ -517,7 +518,7 @@ public void setAwards(List awards) {
     }
 
     public List<IndirectCostRecoveryAccount> getActiveIndirectCostRecoveryAccounts() {
-        List<IndirectCostRecoveryAccount> activeList = new ArrayList<IndirectCostRecoveryAccount>();
+        List<IndirectCostRecoveryAccount> activeList = new ArrayList();
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/org/kuali/kfs/coa/businessobject/Account.java]: Will add

File: src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurchaseOrderValidation.xml (hunk 1 above)
Comment (db483) [reply src/main/resources/edu/cornell/kfs/module/purap/document/validation/configuration/PurchaseOrderValidation.xml]: Good catch

**Comment by db483** [comment]
Chad,

I believe the CsvBatchInputFileTypeBase customization is most probably still needed. I will add that back. Thank you.

File: src/main/resources/edu/cornell/kfs/coa/document/datadictionary/SubObjectCodeGlobalMaintenanceDocument.xml [hunk 4]
```diff
@@ -20,11 +20,9 @@
...
     <property name="maintainableItems">
       <list>
-        <bean parent="MaintainableCollectionDefinition">
-          <property name="name" value="accountGlobalDetails"/>
-          <property name="businessObjectClass" value="org.kuali.kfs.coa.businessobject.AccountGlobalDetail"/>
-          <property name="sourceClassName" value="edu.cornell.kfs.coa.businessobject.AccountGlobalSearch"/>
-          <property name="summaryTitle" value="Account"/>
+        <bean parent="MaintainableCollectionDefinition" p:name="accountGlobalDetails"
+              p:businessObjectClass="org.kuali.kfs.coa.businessobject.AccountGlobalDetail"
+              p:sourceClassName="org.kuali.kfs.coa.businessobject.Account" p:summaryTitle="Account">
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/resources/edu/cornell/kfs/coa/document/datadictionary/SubObjectCodeGlobalMaintenanceDocument.xml]: Thank you, I am updating it.

**Comment by db483** [comment]
This is ready for review again.

**Pull request merged**